### PR TITLE
Add e2e test cases for flow allowlist

### PR DESCRIPTION
**Pull request by IniZio**
ref #4124 / DEV-1219

### CODE REVIEW

File: e2e/pkg/testrunner/runner.go [hunk 1]
```diff
@@ -25,9 +26,14 @@ func NewTestRunner(t *testing.T, path string) *TestRunner {
 func (tr *TestRunner) Run() {
 	var t = tr.T
 
+	err := tr.dumpSchema()
+	if err != nil {
+		t.Fatalf("failed to dump schema: %v", err)
+	}
+
```
Comment (louischan-oursky) [inline e2e/pkg/testrunner/runner.go:33]: Run should not dumpSchema every time it is run.

File: e2e/pkg/testrunner/models.go [hunk 2]
```diff
@@ -1,5 +1,22 @@
...
+	UserImport string              `yaml:"user_import"`
+	CustomSQL  BeforeHookCustomSQL `yaml:"custom_sql"`
+}
+
+var _ = TestCaseSchema.Add("AuthgearYAMLSource", `
+{
+	"type": "object",
+	"additionalProperties": false,
+	"properties": {
+		"extend": { "type": "file-path", "description": "Path to the base authgear.yaml" },
```
Comment (louischan-oursky) [inline e2e/pkg/testrunner/models.go:14]: As far as I know, there is no type called `file-path`.

File: e2e/pkg/testrunner/models.go [hunk 3]
```diff
@@ -12,23 +29,94 @@ const (
 	BeforeHookTypeCustomSQL  BeforeHookType = "custom_sql"
 )
 
+var _ = TestCaseSchema.Add("BeforeHookCustomSQL", `
+{
+	"type": "object",
+	"additionalProperties": false,
+	"properties": {
+		"path": { "type": "string", "format": "file-path", "description": "Path to the custom SQL script" }
```
Comment (louischan-oursky) [inline e2e/pkg/testrunner/models.go:37]: As far as I know, there is no format called `file-path`.

File: e2e/pkg/testrunner/models.go [hunk 4]
```diff
@@ -34,7 +34,7 @@ var _ = TestCaseSchema.Add("BeforeHookCustomSQL", `
 	"type": "object",
 	"additionalProperties": false,
 	"properties": {
-		"path": { "type": "string", "format": "file-path", "description": "Path to the custom SQL script" }
+		"path": { "type": "string", "format": "string", "description": "Path to the custom SQL script" }
```
Comment (louischan-oursky) [inline e2e/pkg/testrunner/models.go:37]: This is no `format: string` neither. Please remove "format".

Comment (IniZio) [reply e2e/pkg/testrunner/models.go]: Removed 🙏

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**